File: 백준/Silver/1912. 연속합/연속합.js
const fs = require('fs');
const input = fs.readFileSync(0, 'utf-8').toString().trim().split('\n');
const N = +input[0];
const nums = input[1].split(' ').map(Number);
const acc = [nums[0]];

for (let i = 1; i < N; i++) {
    const n = nums[i];
    acc[i] = Math.max(acc[i - 1] + n, n);
}

console.log(Math.max(...acc));

File: 백준/Silver/1912. 연속합/연속합.py
N = int(input())
S = list(map(int, input().split()))
dp = [[-float('inf')] * 2] * len(S)
dp[0] = [S[0], S[0]]

for i in range(1,len(S)):
    dp[i] = [max(dp[i-1][0], dp[i-1][1]) + S[i], S[i]]
    
print(max(map(max, dp)))